File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        return solve(numbers, target);
    }
private:
    vector<int> solve(vector<int>& numbers, int target) {
        int i = 0, j = numbers.size() - 1;
        
        while(i < j) {
            int sum = numbers[i] + numbers[j];
            
            if(sum < target) ++i;
            else if(sum > target) --j;
            else break;
        }
        return {i + 1, j + 1};
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i, j = 0, len(numbers) - 1
        
        while i < j:
            _sum = numbers[i] + numbers[j]
            
            if _sum < target: i += 1
            elif _sum > target: j -= 1
            else: break
        
        return [i + 1, j + 1]